modules/dex/src/lib.rs

```diff
@@ -23,7 +24,15 @@ type CurrencyIdOf<T> = <<T as Trait>::Currency as MultiCurrency<<T as system::Tr
 pub trait Trait: system::Trait {
 	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
 	type Currency: MultiCurrencyExtended<Self::AccountId>;
-	type Share: Parameter + Member + AtLeast32Bit + Default + Copy + MaybeSerializeDeserialize;
+	type Share: Into<BalanceOf<Self>>
+		+ From<BalanceOf<Self>>
```

Comment:
`Into<BalanceOf<Self>> + From<BalanceOf<Self>>` basically means it have to be same type of `BalanceOf<Self>`

Do you need both? if not, remove the not used one. if yes, then just `type Share<T> = BalanceOf<T>` in global scope to keep thing simpler.

Reply:
`type Share<T> = BalanceOf<T>` the way will not use the associated type `Share`